### PR TITLE
Add roborazzi.record.filePathStrategy gradle.properties property for dynamic output path resolution

### PR DESCRIPTION
**Pull request by takahirom**
close https://github.com/takahirom/roborazzi/issues/171

### CODE REVIEW

**Comment by takahirom** [comment]
Now, custom_file.png is saved to build/outputs/roborazzi/custom_file.png instead of just custom_file.png. This change has occurred, so we need to exercise caution. Many people rely on the path used by captureRoboImage, so this change could have implications.

**Review by timothyfroehlich — CHANGES_REQUESTED** [review]
I just realized there's this line in the [Readme](https://github.com/takahirom/roborazzi#roborazzirule-optional): `Provide context such as RoborazziOptions and outputDirectoryPath etc for captureRoboImage().`. 

So it's worth pointing out that this is essentially a bug fix to make the behavior of `captureRobo...` match the documentation.

File: include-build/roborazzi-core/src/commonJvmMain/kotlin/com/github/takahirom/roborazzi/fileWithOutputDirectoryContext.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+fun fileWithOutputDirectoryContext(filePath: String): File {
+  if (filePath.startsWith("/")) {
+    return File(filePath)
+  }
+  val outputDirectory = provideRoborazziContext().outputDirectory
+  val outputDirectoryAbsolutePath = File(outputDirectory).absolutePath
+  val filePathAbsolutePath = File(filePath).absolutePath
+  if (filePathAbsolutePath.startsWith("$outputDirectoryAbsolutePath/")) {
+    // Remove duplicated path
+    val prefixRemovedFilePath = filePathAbsolutePath.removePrefix("$outputDirectoryAbsolutePath/")
```
[the remark this answers is not in the document]
Comment (takahirom) [reply include-build/roborazzi-core/src/commonJvmMain/kotlin/com/github/takahirom/roborazzi/fileWithOutputDirectoryContext.kt]: Thanks. Adding a parameter to captureRoboImage() could lead to frequent usage, making it challenging to modify the API later. Additionally, users would be required to specify the property every time. A gradle.properties flag might be a more flexible option.

**Comment by timothyfroehlich** [comment]
Looks good so far! Can you make sure you document the new property in the Readme or somewhere? Thanks!

File: include-build/roborazzi-core/src/commonJvmMain/kotlin/com/github/takahirom/roborazzi/RoborazziOptions.kt [hunk 2]
```diff
@@ -38,6 +39,40 @@ fun roborazziDefaultResizeScale(): Double {
...
+      get() = "RelativePathFromRoborazziContextOutputDirectory"
+  }
+}
+
+
+@ExperimentalRoborazziApi
+fun roborazziCaptureRoboImageFilePathStrategy(): RoborazziCaptureRoboImageFilePathStrategy {
+  return when (
+    System.getProperty(
+      "roborazzi.captureRoboImage.filePathStrategy",
```
Comment (takahirom) [inline include-build/roborazzi-core/src/commonJvmMain/kotlin/com/github/takahirom/roborazzi/RoborazziOptions.kt:62]: This property doesn't affect comparison images like hoge_compare.png. Therefore, I believe roborazzi.record.filePathStrategy would be a more appropriate name.

Comment (takahirom) [reply include-build/roborazzi-core/src/commonJvmMain/kotlin/com/github/takahirom/roborazzi/RoborazziOptions.kt:62]: And we may provide the filePathStrategy in RoborazziContext someday.

File: include-build/roborazzi-core/src/commonJvmMain/kotlin/com/github/takahirom/roborazzi/RoborazziOptions.kt [hunk 3]
```diff
@@ -38,6 +39,40 @@ fun roborazziDefaultResizeScale(): Double {
...
+      get() = "RelativePathFromRoborazziContextOutputDirectory"
+  }
+}
+
+
+@ExperimentalRoborazziApi
+fun roborazziRecordFilePathStrategy(): RoborazziRecordFilePathStrategy {
+  return when (
+    System.getProperty(
+      "roborazzi.record.filePathStrategy",
```
Comment (takahirom) [inline include-build/roborazzi-core/src/commonJvmMain/kotlin/com/github/takahirom/roborazzi/RoborazziOptions.kt:62]: The property name has been changed.
https://github.com/takahirom/roborazzi/pull/172#discussion_r1339384610

**Comment by takahirom** [comment]
@timothyfroehlich 
Thank you for your review! I've uploaded the document. Could you please take a look?
https://github.com/takahirom/roborazzi/blob/bd771a41f3d82ec788366291ca3fef4da313b18a/README.md#roborazzi-gradleproperties-options

**Pull request merged**